### PR TITLE
fixed hyperventing agent 3.4

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1456/

File: arangod/Agency/Agent.cpp [hunk 1]
```diff
@@ -1775,15 +1775,18 @@ query_t Agent::gossip(query_t const& in, bool isCallback, size_t version) {
...
   if (!_config.poolComplete() || _config.matchPeer(id, endpoint)) {
-      
+
+    auto before = _config.version();
     if (!_config.upsertPool(pslice, id)) {
       LOG_TOPIC(FATAL, Logger::AGENCY) << "Discrepancy in agent pool!";
       FATAL_ERROR_EXIT();      /// disagreement over pool membership are fatal!
     }
     auto pool = _config.pool();
+    changed = before < _config.version();
```
Comment (matthewvon) [inline arangod/Agency/Agent.cpp]: What causes _config.version() to change between line 1783 and line 1789?

Comment (kvahed) [reply arangod/Agency/Agent.cpp]: _config.upsertPool(pslice, id)

Comment (kvahed) [reply arangod/Agency/Agent.cpp]: This is actually, what prevents the craziness now going forward.

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1515/

**Pull request merged**